src/components/Missions/Missions.js

```diff
@@ -1,12 +1,21 @@
 import { React, useEffect } from 'react';
 import { useDispatch, useSelector } from 'react-redux';
-import MissionsItem from './MissionsPage';
+import MissionsPage from './MissionsPage';
 import './Missions.module.css';
-import { getMissions } from '../../redux/missions/missions';
+import { getMissions, joinMissions, leaveMissions } from '../../redux/missions/missions';
 
 const Mission = () => {
   const dispatch = useDispatch();
   const missionData = useSelector((state) => state.missions);
+
+  const joinMissionHandler = (id) => {
+    dispatch(joinMissions(id));
+  };
+
+  const leaveMissionHandler = (id) => {
+    dispatch(leaveMissions(id));
+  };
+
   useEffect(() => {
     if (missionData.length === 0) {
       dispatch(getMissions());
```

Comment:
This use Effect is no longer necesary to dispatch the getMissions(); you can comment this lines because we are dispatching this only once in the app.js, and this is now visible for your useSelector, try this and see what happens